### ISSUE TITLE
Validator: `validatePropertyConstraints` only checks the immediate `properties` of (1) the root schema and (2) each `components/schemas/*`

### ISSUE THREAD

**Issue by simihablo** [meshery/schemas#686]
## Description

`validatePropertyConstraints` only checks the immediate `properties` of (1) the root schema and (2) each `components/schemas/*`. It will *not* validate nested object schemas (e.g., `properties.*.properties`), array item schemas (`properties.*.items.properties`), or schemas defined under combiners (`allOf`/`oneOf`/`anyOf`). This creates false negatives (validator can report “0 advisory issues” while nested properties still violate Rules 36–39). Refactor this to use a recursive schema walker that visits all sub-schemas (properties, items, additionalProperties, and combiner branches), and apply Rules 36–39 to every object schema’s `properties`.

```javascript
/**
 * Recursively walk a schema and apply property-level constraints (Rules 36–39)
 * to every object schema that declares `properties`.
 *
 * @param {string} filePath    The file being validated (for error context).
 * @param {object} schema      The current schema node.
 * @param {string|null} schemaName Logical name/path for the schema (for messages).
 */
function walkSchemaForPropertyConstraints(filePath, schema, schemaName) {
  if (!schema || typeof schema !== "object") return;

  // Apply Rules 36–39 to this schema's own properties if it is an object.
  if (schema.type === "object" && schema.properties && typeof schema.properties === "object") {
    validatePropertyDescriptions(filePath, schema.properties, schemaName || null);
    validateStringConstraints(filePath, schema.properties, schemaName || null);
    validateNumericBounds(filePath, schema.properties, schemaName || null);
    validateIdFormat(filePath, schema.properties, schemaName || null);
  }

  // Recurse into nested property schemas.
  if (schema.properties && typeof schema.properties === "object") {
    for (const [propName, propSchema] of Object.entries(schema.properties)) {
      if (!propSchema || typeof propSchema !== "object") continue;
      const childName = schemaName ? `${schemaName}.${propName}` : propName;
      walkSchemaForPropertyConstraints(filePath, propSchema, childName);
    }
  }

  // Recurse into array item schemas.
  if (schema.items && typeof schema.items === "object") {
    const childName = schemaName ? `${schemaName}[]` : null;
    walkSchemaForPropertyConstraints(filePath, schema.items, childName);
  }

  // Recurse into map value schemas defined via additionalProperties.
  if (schema.additionalProperties && typeof schema.additionalProperties === "object") {
    const childName = schemaName ? `${schemaName}.*` : null;
    walkSchemaForPropertyConstraints(filePath, schema.additionalProperties, childName);
  }

  // Recurse into combiner branches (allOf / oneOf / anyOf).
  for (const combiner of ["allOf", "oneOf", "anyOf"]) {
    const branch = schema[combiner];
    if (!Array.isArray(branch)) continue;
    for (let i = 0; i < branch.length; i += 1) {
      const subSchema = branch[i];
      if (!subSchema || typeof subSchema !== "object") continue;
      const childName =
        schemaName != null ? `${schemaName}.${combiner}[${i}]` : `${combiner}[${i}]`;
      walkSchemaForPropertyConstraints(filePath, subSchema, childName);
    }
  }
}

function validatePropertyConstraints(filePath, doc) {
  // Entity-level schemas (top-level properties in .yaml files)
  if (doc && typeof doc === "object" && doc.type === "object" && doc.properties) {
    walkSchemaForPropertyConstraints(filePath, doc, null);
  }

  // Component schemas in api.yml
  if (doc && typeof doc === "object" && doc.components?.schemas) {
    for (const [schemaName, schema] of Object.entries(doc.components.schemas)) {
      if (!schema || typeof schema !== "object") continue;
      walkSchemaForPropertyConstraints(filePath, schema, schemaName);
```

_Originally posted by @Copilot in https://github.com/meshery/schemas/pull/682#discussion_r3017206139_

## Feedback

Good suggestion for a follow-up. The current implementation covers immediate `properties` blocks at root and `components/schemas/*` level, which catches the vast majority of violations. Recursive descent into nested object properties can be added as a separate enhancement (it risks false positives on inline `$ref` compositions and `oneOf`/`allOf` branches that need special handling).

_Originally posted by @leecalcote in https://github.com/meshery/schemas/pull/682#discussion_r3017294105_

**Comment by PARTH-TUSSLE** [comment]
@simihablo I would like to work on this issue, could you assign it to me?
